### PR TITLE
Update release plugin for sonatype

### PR DESCRIPTION
**Pull request by Siedlerchr**
TODO: add creds to github secrets etc

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,12 +3,12 @@ plugins {
...
         id 'signing'
         id 'org.openjfx.javafxplugin' version '0.1.0'
-        id "io.codearte.nexus-staging" version "0.30.0"
+        id "io.github.gradle-nexus.publish-plugin" version "1.3.0"
 }
 
 allprojects {
     group = 'org.jabref'
-    version = '2.0.0-SNAPSHOT'
+    version = '2.0.0'
```
Comment (koppor) [inline build.gradle:11]: Keep SNAPSHOT as main is the development branch.

File: build.gradle [hunk 2]
```diff
@@ -92,12 +92,14 @@ publishing {
...
+
             credentials {
-                username = System.getenv("MAVEN_USERNAME")
-                password = System.getenv("MAVEN_PASSWORD")
+                username = ossrhUsername
+                password = ossrhPassword
             }
         }
-        // This currently only works for released versions, as github has problems with overwriting snapshots
+       /* // This currently only works for released versions, as github has problems with overwriting snapshots
```
Comment (koppor) [inline build.gradle:102]: Can we try to craft GitHub releases? (Benefit?)

File: build.gradle [hunk 3]
```diff
@@ -116,13 +118,13 @@ signing {
...
+            username = ossrhUsername
+            password = ossrhPassword
+            nexusUrl.set(uri("https://s01.oss.sonatype.org/service/local/"))
+            snapshotRepositoryUrl.set(uri("https://s01.oss.sonatype.org/content/repositories/snapshots/"))
+        }
+    }
 }
-
-// Don't publish snapshots to Maven Central
-closeAndReleaseRepository.onlyIf { isReleaseVersion }
```
Comment (koppor) [inline build.gradle:126]: Shouldn't we do similar things for Nexus publishing (to publish releases only)

**Review by koppor — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (Siedlerchr) [reply build.gradle:126]: could be possible, just the tasks having different names

**Comment by koppor** [comment]
` > Could not get unknown property 'ossrhUsername' for Credentials [username: null] of type org.gradle.internal.credentials.DefaultPasswordCredentials_Decorated.`

@Siedlerchr Could you have a look? (Sorry for not having followed up earlier on this)

File: build.gradle (hunk 2 above)
Comment (Siedlerchr) [reply build.gradle:102]: I don't know where this comment came from

**Comment by Siedlerchr** [comment]
I will try to take a look at the next days... 
The above state was for doing a release locally.

**Comment by koppor** [comment]
Background why we dived into this one: We think, we need to rewrite the part of FXML loading due to https://github.com/oracle/graal/issues/7682, which requires to use `Stream` instead of `Path` for a "clean" loading from files stored inside an application distribution.

**Review by koppor — APPROVED** [review]

**Pull request merged**